treeherder/model/management/commands/cycle_data.py

```diff
@@ -55,32 +72,65 @@ def handle(self, *args, **options):
 
         self.debug("cycle interval... {}".format(cycle_interval))
 
-        for repository in Repository.objects.all():
-            self.debug("Cycling repository: {0}".format(repository.name))
-            rs_deleted = Job.objects.cycle_data(repository,
-                                                cycle_interval,
-                                                options['chunk_size'],
-                                                options['sleep_time'])
-            self.debug("Deleted {} jobs from {}".format(rs_deleted,
-                                                        repository.name))
-
-            # TODO: Fix the performance issues and re-enable:
-            # https://bugzilla.mozilla.org/show_bug.cgi?id=1346567#c10
-            if False and repository.expire_performance_data:
-                PerformanceDatum.objects.cycle_data(repository,
+        if options['target'] in ["all", "jobs"]:
+            for repository in Repository.objects.all():
+                self.debug("Cycling repository: {0}".format(repository.name))
+                rs_deleted = Job.objects.cycle_data(repository,
                                                     cycle_interval,
                                                     options['chunk_size'],
                                                     options['sleep_time'])
+                self.debug("Deleted {} jobs from {}".format(rs_deleted,
+                                                            repository.name))
+
+                # TODO: Fix the performance issues and re-enable:
+                # https://bugzilla.mozilla.org/show_bug.cgi?id=1346567#c10
+                if False and repository.expire_performance_data:
+                    PerformanceDatum.objects.cycle_data(repository,
+                                                        cycle_interval,
+                                                        options['chunk_size'],
+                                                        options['sleep_time'])
+
+        if options['target'] in ["all", "logdata"]:
+            self.cycle_expired_log_records(
+                options['chunk_size'],
+                options['sleep_time'],
+                options['iterations'],
+                cycle_interval)
+
+        if options['target'] in ["all", "refdata"]:
+            self.cycle_non_job_data(
+                options['chunk_size'],
+                options['sleep_time'],
+                cycle_interval)
+
+    def cycle_expired_log_records(self, chunk_size, sleep_time, iterations, cycle_interval):
+        self.debug("delete expired FailureLines")
+        for x in range(0, iterations):
```

Comment:
I think maybe we can remove the iterations here and instead run until completion? (Now I've run the manual purge, there will be much less to delete)